File: Regular/384. Shuffle an Array/Solution.js
/**
 * @param {number[]} nums
 */
var Solution = function(nums) {
    this.original = nums;
    this.rand = Array.from(nums);
};

/**
 * Resets the array to its original configuration and return it.
 * @return {number[]}
 */
Solution.prototype.reset = function() {
    return this.original;
};

/**
 * Returns a random shuffling of the array.
 * @return {number[]}
 */
Solution.prototype.shuffle = function() {
    return this.rand.sort((a, b) => Math.random() - 0.5);
};

/** 
 * Your Solution object will be instantiated and called as such:
 * var obj = new Solution(nums)
 * var param_1 = obj.reset()
 * var param_2 = obj.shuffle()
 */

File: Regular/384. Shuffle an Array/Solution.java
class Solution {
    private int[] original;
    private Integer[] rand;

    public Solution(int[] nums) {
        rand = new Integer[nums.length];
        for (int i = 0; i < nums.length; i++) rand[i] = nums[i];
        original = nums;
    }
    
    /** Resets the array to its original configuration and return it. */
    public int[] reset() {
        return original;
    }
    
    /** Returns a random shuffling of the array. */
    public int[] shuffle() {
        Arrays.sort(rand, (a, b) -> Math.random() >= 0.5 ? 1 : -1);
        int[] out = new int[rand.length];
        for (int i = 0; i < rand.length; i++) out[i] = rand[i];
        return out;
    }
}

/**
 * Your Solution object will be instantiated and called as such:
 * Solution obj = new Solution(nums);
 * int[] param_1 = obj.reset();
 * int[] param_2 = obj.shuffle();
 */
